### PR TITLE
fix: update obs sdk and obs sdk support roundTripper

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:
update obs sdk and obs sdk support roundTripper.
add user-agent
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/obs' TESTARGS='-run TestAccObsBucketObjectDataSource_content'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/obs -v -run TestAccObsBucketObjectDataSource_content -timeout 360m -parallel 4 
=== RUN   TestAccObsBucketObjectDataSource_content 
=== PAUSE TestAccObsBucketObjectDataSource_content
=== CONT  TestAccObsBucketObjectDataSource_content
--- PASS: TestAccObsBucketObjectDataSource_content (15.47s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/obs       15.523s

make testacc TEST='./huaweicloud/services/acceptance/obs' TESTARGS='-run TestAccDataSourceObsBuckets_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/obs -v -run TestAccDataSourceObsBuckets_basic -timeout 360m -parallel 4 
=== RUN   TestAccDataSourceObsBuckets_basic 
=== PAUSE TestAccDataSourceObsBuckets_basic
=== CONT  TestAccDataSourceObsBuckets_basic
--- PASS: TestAccDataSourceObsBuckets_basic (9.69s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/obs       9.741s

make testacc TEST='./huaweicloud/services/acceptance/obs' TESTARGS='-run TestAccObsBucketObject_source'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/obs -v -run TestAccObsBucketObject_source -timeout 360m -parallel 4 
=== RUN   TestAccObsBucketObject_source 
=== PAUSE TestAccObsBucketObject_source
=== CONT  TestAccObsBucketObject_source
--- PASS: TestAccObsBucketObject_source (15.52s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/obs       15.573s

make testacc TEST='./huaweicloud/services/acceptance/obs' TESTARGS='-run TestAccObsBucketPolicy_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/obs -v -run TestAccObsBucketPolicy_basic -timeout 360m -parallel 4 
=== RUN   TestAccObsBucketPolicy_basic 
=== PAUSE TestAccObsBucketPolicy_basic
=== CONT  TestAccObsBucketPolicy_basic
--- PASS: TestAccObsBucketPolicy_basic (9.58s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/obs       9.623s

make testacc TEST='./huaweicloud/services/acceptance/obs' TESTARGS='-run TestAccObsBucket_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/obs -v -run TestAccObsBucket_basic -timeout 360m -parallel 4 
=== RUN   TestAccObsBucket_basic 
=== PAUSE TestAccObsBucket_basic
=== CONT  TestAccObsBucket_basic
--- PASS: TestAccObsBucket_basic (14.83s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/obs       14.875s

```

### CODE REVIEW

File: huaweicloud/config/config.go [hunk 1]
```diff
@@ -179,12 +180,16 @@ func (c *Config) ObjectStorageClientWithSignature(region string) (*obs.ObsClient
 		}
 	}
 
+	httpClient := c.DomainClient.HTTPClient
+	clientConfigure := obs.WithHttpClient(&httpClient)
+	userAgentConfigure := obs.WithUserAgent(userAgent)
```
Comment (ShiChangkuo) [inline huaweicloud/config/config.go]: custom defined user agent should also be supported

Comment (deer-hang) [reply huaweicloud/config/config.go]: done, customUserAgent has been supported

**Pull request merged**